server/src/database/migrations/sql/8_add_analytics_board_deletion_trigger.up.sql

```diff
@@ -0,0 +1,62 @@
+CREATE TABLE IF NOT EXISTS deleted_boards(
+    id uuid not null primary key,
+    access_policy access_policy,
+    total_columns int, 
+    hidden_columns int,
+    total_notes int,
+    hidden_notes int,
+    first_note_created timestamptz,
+    last_note_created timestamptz,
+    avg_chars_per_note int,
+    total_users int,
+    total_admins int,
+    total_votes int, 
+    total_votings int, 
+    created_at timestamptz,
+    deleted_at timestamptz default now()
+);
+
+CREATE OR REPLACE FUNCTION record_deleted_board_statistics()
+    RETURNS TRIGGER AS $$
+    DECLARE 
+        total_columns int;
+        hidden_columns int;
+        total_notes int;
+        hidden_notes int;
+        first_note_created timestamptz;
+        last_note_created timestamptz;
+        avg_chars_per_note int;
+        total_users int;
+        total_admins int;
+        total_votes int;
+        total_votings int;
+
+        BEGIN 
+            -- Retrieve all needed stats
+            SELECT count(id) INTO total_columns FROM columns WHERE board = OLD.id;
+            SELECT count(id) INTO hidden_columns FROM columns c WHERE board = OLD.id AND c."visible" = false;
+
+            SELECT count(id) INTO total_notes FROM notes WHERE board = OLD.id;
+            SELECT count(n.id) INTO hidden_notes FROM notes n INNER JOIN columns c ON n."column" = c.id WHERE c."visible" = false AND n."board" = OLD.id;
+            SELECT created_at INTO first_note_created FROM notes WHERE board = OLD.id ORDER BY created_at LIMIT 1;
```

Comment:
Maybe select min/max instead? Dunno whether that's faster than order by

Reply:
TL;DR: min/max is simpler to read. Performance is not noticeably different in local setup.

I played around with `explain` & `analyze` in order to get a deeper look into the query. 
The output, as I understand it, is that the min/max creates less overhead. In terms of execution time, I could not spot a noticeable difference (both bouncing in the same time range). This might differ from the actual prod-system.

Here is the output of the analyzed queries:

**Order By:**
![orderby png ](https://github.com/inovex/scrumlr.io/assets/56362368/a567ff80-5c02-49cc-a1ac-c55455c43d6f)


**Min/Max:**
![min](https://github.com/inovex/scrumlr.io/assets/56362368/79d35f09-db25-4b96-8f83-1341243c7da9)